File: Data Structures & Algorithms/longest-consecutive-sequence/submission-8.rs
impl Solution {
    pub fn longest_consecutive(nums: Vec<i32>) -> i32 {
        let set:HashSet<i32>=nums.iter().cloned().collect();
        let mut max=0;
        for mut num in nums{
            if !set.contains(&(num-1)){
                let mut count:usize=0;
                while set.contains(&num){
                    count+=1;
                    num+=1;
                }
                if max<count{
                    max=count;
                }
            }
        }
        return max as i32;
    }
}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-8.java
class Solution {
    public int maxProfit(int[] prices) {
        int len=prices.length;
        int l=0;
        int max=0;
        for(int r=0; r<len; r++){
            if(prices[r]<prices[l]){
                l=r;
            }else{
                max=Math.max(max, prices[r]-prices[l]);
            }
        }
        return max;
    }
}
